portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,10 +28,10 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    ArrayList<Comment> comments; 
+    List comments; 
 
     public DataServlet() {
-        this.comments = new ArrayList<Comment>();
+        this.comments = new ArrayList<>();
```

Comment:
consider overriding the init method instead of doing this in the constructor. See dalen's PR for an example.